readthedocs/redirects/querysets.py

```diff
@@ -35,16 +36,52 @@ def api(self, user=None):
             queryset = self._add_from_user_projects(queryset, user)
         return queryset
 
-    def get_redirect_path_with_status(self, path, full_path=None, language=None, version_slug=None):
+    def get_redirect_path_with_status(
+        self, path, full_path=None, language=None, version_slug=None, forced_only=False
+    ):
+        """
+        Get the final redirect with its status code.
+
+        :param path: Is the path without the language and version parts.
+        :param full_path: Is the full path including the language and version parts.
+        :param forced_only: Include only forced redirects in the results.
+        """
+
+        # Small optimization to skip executing the big query below.
+        if forced_only and not self.filter(force=True).exists():
+            return None, None
+
+        def normalize_path(path):
```

Comment:
This inline function feels a bit weird.. Is there a reason it isn't in a utils file?

Reply:
it was very specific to this method, I'll move it to be a private method